src/components/SignatureRequest.js

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { Button, GU, textStyle, useTheme } from '@aragon/ui'
+import { useWallet } from '../providers/Wallet'
+import { signMessage } from '../lib/web3-utils'
+import { dayjs } from '../utils/date-utils'
+import signRequestIllustration from '../../src/assets/signRequest.svg'
+import signRequestSuccessIllustration from '../../src/assets/signRequestSuccess.svg'
+import signRequestFailIllustration from '../../src/assets/signRequestFail.svg'
+
+const SignerRequest = React.memo(function SignerRequest({
+  actionText,
+  compactMode,
+  onActionErrorScreen,
+  onActionSuccessScreen,
+  onSignSuccessAction,
```

Comment:
Yes, that is something that i have been thinking to change because i am repeating this pattern in another components where might worth just to handle everything on the parent